espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt

```diff
@@ -0,0 +1,303 @@
+        val adapter = adapterView.adapter ?: return
+        val adapterCount = adapter.count
+        val adapterData = ArrayList<String>()
+        var isAdapterTypeSet = false
+        for (i in 0 until adapterCount) {
+            val adapterItem = adapter.getItem(i) ?: continue
+            adapterData.add(adapterItem.toString())
+
+            // Get the type of the adapter item
+            if (!isAdapterTypeSet) {
+                setAttribute(ViewAttributesEnum.ADAPTER_TYPE, adapterItem.javaClass.simpleName)
+                isAdapterTypeSet = true
+            }
+        }
+        if (!adapterData.isEmpty()) {
+            setAttribute(ViewAttributesEnum.ADAPTERS, TextUtils.join(",", adapterData))
+        }
+    }
+
+    /**
+     * Recursively visit all of the views and map them to XML elements
+     *
+     * @param view  The root view
+     * @param depth The current traversal depth
+     */
+    @Throws(IOException::class)
+    private fun serializeView(view: View?, depth: Int) {
+        if (view == null) {
+            return
+        }
+
+        val viewElement = ViewElement(view)
+        val tagName = toXmlNodeName(viewElement.className)
+        serializer?.startTag(NAMESPACE, tagName)
+
+        // Set attributes
+        setAttribute(ViewAttributesEnum.INDEX, viewElement.index)
+        setAttribute(ViewAttributesEnum.PACKAGE, viewElement.packageName)
+        setAttribute(ViewAttributesEnum.CLASS, viewElement.className)
+        setAttribute(ViewAttributesEnum.CONTENT_DESC, viewElement.contentDescription)
+        setAttribute(ViewAttributesEnum.CHECKABLE, viewElement.isCheckable)
+        setAttribute(ViewAttributesEnum.CHECKED, viewElement.isChecked)
+        setAttribute(ViewAttributesEnum.CLICKABLE, viewElement.isClickable)
+        setAttribute(ViewAttributesEnum.ENABLED, viewElement.isEnabled)
+        setAttribute(ViewAttributesEnum.FOCUSABLE, viewElement.isFocusable)
+        setAttribute(ViewAttributesEnum.FOCUSED, viewElement.isFocused)
+        setAttribute(ViewAttributesEnum.SCROLLABLE, viewElement.isScrollable)
+        setAttribute(ViewAttributesEnum.LONG_CLICKABLE, viewElement.isLongClickable)
+        setAttribute(ViewAttributesEnum.PASSWORD, viewElement.isPassword)
+        setAttribute(ViewAttributesEnum.SELECTED, viewElement.isSelected)
+        setAttribute(ViewAttributesEnum.VISIBLE, viewElement.isVisible)
+        setAttribute(ViewAttributesEnum.BOUNDS, viewElement.bounds.toShortString())
+        val viewText = viewElement.text
+        if (viewText != null) {
+            setAttribute(ViewAttributesEnum.TEXT, viewText.rawText)
+            setAttribute(ViewAttributesEnum.HINT, viewText.isHint)
+        }
+        setAttribute(ViewAttributesEnum.RESOURCE_ID, viewElement.resourceId)
+        setAttribute(ViewAttributesEnum.VIEW_TAG, viewElement.viewTag)
+        if (view is AdapterView<*>) {
+            recordAdapterViewInfo((view as AdapterView<*>?)!!)
```

Comment:
I thought Kotlin would understand that view is of type of AdaperView automatically